### ISSUE TITLE
Is the key / keyname needed. Would you consider not making the keyname mandatory

### ISSUE THREAD

**Issue by tbenade** [buildkite/elastic-ci-stack-for-aws#413]
At first glance EC2 key pair does not look critical to the stack. Could it be done away with as a mandatory option?

Rationale
We aspire to and advocate no ssh keys in production. CI/CD systems given the capability it normally has in the AWS account is definitely considered to be a production system in our world.

Secondly we find having no access to the instances avoids any unrepeatable configuration tom foolery and promotes automation over clickops.

Thanks

**Comment by lox** [comment]
Yup, I've been meaning to make it optional for ages! Would happily accept a PR.

**Comment by tbenade** [comment]
Had a decent go at this but unfortunately cant see a way to make parameter type of AWS::EC2::KeyPair::KeyName optional. If it was plain old string we could do it but cant be sure that would not constitute a breaking change for someone. Any suggestions?

**Comment by lox** [comment]
This is a tricky one. I think I'm probably ok with converting it to a string field @tbenade, PR away and we'll do some experimenting.

**Comment by zsims** [comment]
I'll look at a PR for this, according to AWS about changing the type:

> As long as the finally resolved value of the parameter is the same, no update will happen

As you suspected there isn't a way to have optional AWS typed parameters, as they're all resolved up front.

**Comment by zsims** [comment]
@lox @tbenade PR up: https://github.com/buildkite/elastic-ci-stack-for-aws/pull/444
